src/test/java/uk/gov/companieshouse/siccode/api/search/CombinedSicActivitiesRepositoryTest.java

```diff
@@ -0,0 +1,119 @@
+
+	@Autowired
+    private CombinedSicActivitiesRepository combinedSicActivitiesRepository;
+
+	@Autowired
+	private MongoTemplate mongoTemplate;
+
+	private final static CombinedSicActivitiesStorageModel BARLEY_FARMING = new CombinedSicActivitiesStorageModel("10", "01110", "Barley Farming",
+	"barley farming", "Growing of cereals except rice, leguminous crops and oil seeds", false);
+
+	private final static CombinedSicActivitiesStorageModel BARLEY_GROWING = new CombinedSicActivitiesStorageModel("15", "01110", "Barley growing",
+	"barley growing", "Growing of cereals except rice, leguminous crops and oil seeds", false);
+
+	private final static CombinedSicActivitiesStorageModel BEAN_GROWING_ORGANIC = new CombinedSicActivitiesStorageModel("17", "01110", "Bean growing organic",
+	"bean growing organic", "Growing of cereals except rice, leguminous crops and oil seeds and manufacture", false);
+
+	private final static CombinedSicActivitiesStorageModel BEAN_GROWING = new CombinedSicActivitiesStorageModel("18", "01110", "Bean growing",
+	"bean growing", "Growing of cereals except rice, leguminous crops and oil seeds", false);
+
+	private final static CombinedSicActivitiesStorageModel ARMOURED_CAR_SERVICES = new CombinedSicActivitiesStorageModel("20", "80100", "Armoured car services",
+	"armoured car services", "Private security activities", false);
+
+	private final static CombinedSicActivitiesStorageModel BARLEY_MALTING = new CombinedSicActivitiesStorageModel("30", "11060", "Barley malting (manufacture)",
+	"barley malting manufacture", "Manufacture of malt", false);
+
+	private final static CombinedSicActivitiesStorageModel BUS_MANUFACTURE = new CombinedSicActivitiesStorageModel("40", "29201", "Body for bus (manufacture)",
+	"body for bus manufacture", "Manufacture of bodies (coachwork) for motor vehicles (except caravans)", false);
+
+	@BeforeEach
+	public void beforeEach() {
+	
+		var combinedSicActivities = Arrays.asList(BARLEY_FARMING, BARLEY_GROWING, BEAN_GROWING, BEAN_GROWING_ORGANIC, ARMOURED_CAR_SERVICES, BARLEY_MALTING, BUS_MANUFACTURE);
+		combinedSicActivitiesRepository.saveAll(combinedSicActivities);	
+
+		// We need to create a text index for this collection so that a Text Search works (else get the error - failed with error code 27 and error message ‘text index required for $text query")
+		mongoTemplate.indexOps("combined_sic_activities").ensureIndex(TextIndexDefinition.builder().onFields("activity_description_search_field").build());
+	}
+
+	@AfterEach
+	public void afterEach() {
+		mongoTemplate.dropCollection(CombinedSicActivitiesStorageModel.class);
+	}
+
+
+	@Test
+	void textSearchThreeItemsAnyMatchVerifySorting() {
+
+		var results = combinedSicActivitiesRepository.findAllByOrderByScore(
+			new SicCodeSearchTextCriteria("Bean Growing Organic").getTextCriteriaMatchAny());
+
+		assertEquals(3, results.size());
+
+		assertThat(results, contains( BEAN_GROWING_ORGANIC, BEAN_GROWING, BARLEY_GROWING));
```

Comment:
Why is this on contains, and the next containsInAnyOrder?

Reply:
With contains it checks the order (and this test checks the ordering of the results)